package/kedro_viz/__init__.py

```diff
@@ -1,3 +1,17 @@
 """Kedro plugin for visualising a Kedro pipeline"""
+import sys
+import warnings
 
 __version__ = "6.4.0"
+
+
+class KedroVizPythonVersionWarning(UserWarning):
+    """Custom class for warnings about incompatibilities with Python versions."""
+
+
+if sys.version_info >= (3, 11):
```

Comment:
If we aim to declare official 3.11 compatibility with this set of releases (Kedro 0.18.13, kedro-datasets 1.6.0, Kedro-Viz 6.5.0 or whatever comes next), this warning should be for the next CPython version:

```suggestion
if sys.version_info >= (3, 12):
```

Notice that Kedro 0.18.12 still emits this warning for Python 3.11, but https://github.com/kedro-org/kedro/pull/2851 bumped this to 3.12.